app/kubemci/cmd/remove_clusters_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/kubeutils"
+)
+
+// Test to verify validate.
+// This tests all flags except `--gcp-project`, which is tested by the next test.
+func TestValidateRemoveClustersArgs(t *testing.T) {
+	// validateRemoveClustersArgs should return an error with empty options.
+	options := removeClustersOptions{}
+	if err := validateRemoveClustersArgs(&options, []string{}); err == nil {
+		t.Errorf("Expected error for emtpy options")
+	}
+
+	// validateRemoveClustersArgs should return an error with missing load balancer name.
+	options = removeClustersOptions{
+		IngressFilename:    "ingress.yaml",
+		GCPProject:         "gcp-project",
+		KubeconfigFilename: "kubeconfig",
+	}
+	if err := validateRemoveClustersArgs(&options, []string{}); err == nil {
+		t.Errorf("Expected error for missing load balancer name")
+	}
+
+	// validateRemoveClustersArgs should return an error with missing ingress.
+	options = removeClustersOptions{
+		GCPProject:         "gcp-project",
+		KubeconfigFilename: "kubeconfig",
+	}
+	if err := validateRemoveClustersArgs(&options, []string{"lbname"}); err == nil {
+		t.Errorf("Expected error for missing ingress")
+	}
+
+	// validateRemoveClustersArgs should return an error with missing kubeconfig.
+	options = removeClustersOptions{
+		IngressFilename: "ingress.yaml",
+		GCPProject:      "gcp-project",
+	}
+	if err := validateRemoveClustersArgs(&options, []string{"lbname"}); err == nil {
```

Comment:
In all the 4 cases above, don't you also want to check if the expected error was returned? It's fine if you do this in a follow up CL, but could you add a TODO in that case?

Also, won't this test be more readable and less tedious if you move all the test case definitions to the beginning like the way you do in url maps tests? Something like:

```go
testCases := []struct{
    opts removeClustersOptions
    args []string
    expectedErr error
    failMsg string
}{
    {
        opts: removeClustersOptions{},
        args: []string{},
        expectedErr: errors.New("Some error"),
        failMsg: "Expected error for emtpy options",
    },
    {
        opts: removeClustersOptions{
		IngressFilename:    "ingress.yaml",
		GCPProject:         "gcp-project",
		KubeconfigFilename: "kubeconfig",
	},
        args: []string{"lbname"},
        expectedErr: errors.New("Some other error"),
        failMsg: "Expected error for missing load balancer name",
    },
    ...
}

for i, case := range testCases {
    if err := validateRemoveClustersArgs(&case.opts, opts.args); err == case.expectedErr {
        t.Errorf(case.failMsg)
    }
}

```